src/components/CodeAssignmentModal/index.jsx

```diff
@@ -351,7 +354,21 @@ class BaseCodeAssignmentModal extends React.Component {
       options.codes = [code.code];
     }
 
-    return sendCodeAssignment(couponId, options)
+    let pendingEnterpriseUserData;
+    if (hasTextAreaEmails) {
+      pendingEnterpriseUserData = validEmails.map((email) => ({
+        user_email: email,
+        enterprise_customer: enterpriseUuid,
+      }));
+    } else {
+      pendingEnterpriseUserData = {
+        user_email: formData['email-address'],
```

Comment:
Is this else needed? Where is this from if not the text area or csv (as we put either in the validEmails var)?
Actually I can't even find the email-address field in this file, so I'm extra confused.

Reply:
So, there are two different forms (but for reasons they are the same ReactForm? this is an architectural problem, but not one I'm dealing with now). One has a single form field for an email address; you get to it by clicking the assign link in the table, and the other is for bulk assignment, and that's the text area. So if we're bulk assigning, we need one option, if not, we use this. It confused me, too.

Reply:
The `formData` is from redux-form, I used to know more about how it worked, but basically it creates the formData object when you hit submit and passes that data to this function.